src/tags/for.ts

```diff
@@ -66,8 +67,14 @@ export default class extends Tag {
     }, collection)
 
     ctx.setRegister(continueKey, (hash['offset'] || 0) + collection.length)
-    const scope = { forloop: new ForloopDrop(collection.length, this.collection.getText(), this.variable) }
+
+    const forloopRegister = ctx.getRegister('forloops', [])
+    const parentloop = forloopRegister[forloopRegister.length - 1] ?? NULL
```

Comment:
since `ForloopDrop` constructor already handles a `undefined` parent, here we don't need to default it to `NULL`, let it be undefined will be fine I guess.